lib/resty/upstream/healthcheck.lua

```diff
@@ -597,6 +605,16 @@ function _M.spawn_checker(opts)
         return nil, "no upstream specified"
     end
 
+    local hook_down = opts.hook_down
```

Comment:
While I agree with your premise, I think following the existing style of the module here is important, and concern about trying to second-guess the JIT compiler is not a huge priority because this is only called once in `init_worker`.

Might be worth a separate cleanup commit at some point?